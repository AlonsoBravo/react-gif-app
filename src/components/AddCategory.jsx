import { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ onAddCategory }) => {
	const [inputValue, setInputValue] = useState('');

	const onInputChange = ({ target }) => {
		setInputValue(target.value);
	};

	const onSubmit = (event) => {
		event.preventDefault();

		const value = inputValue.trim();

		if (value.length <= 1) {
			return;
		}

		onAddCategory(value);
		setInputValue('');
	};

	return (
		<form onSubmit={onSubmit} aria-label='form'>
			<input
				type={'text'}
				placeholder='Buscar gifs'
				value={inputValue}
				onChange={onInputChange}
			/>
		</form>
	);
};

AddCategory.propTypes = {
	onAddCategory: PropTypes.func.isRequired,
};

export default AddCategory;
