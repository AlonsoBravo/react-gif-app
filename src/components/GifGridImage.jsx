import PropTypes from 'prop-types';

const GifGridImage = ({ url, title }) => {
	return (
		<div className='card'>
			<img src={url} alt={title} />
			<p>{title}</p>
		</div>
	);
};

GifGridImage.propTypes = {
	url: PropTypes.string.isRequired,
	title: PropTypes.string.isRequired,
};

export default GifGridImage;
