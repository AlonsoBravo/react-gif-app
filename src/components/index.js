export { default as AddCategory } from './AddCategory.jsx';
export { default as GifGrid } from './GifGrid.jsx';
export { default as GifGridImage } from './GifGridImage.jsx';
