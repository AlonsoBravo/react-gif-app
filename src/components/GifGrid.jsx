import { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import GifGridImage from './GifGridImage';
import { useFetchGifs } from '../hooks';

const GifGrid = ({ category }) => {
	const { gifs, isLoading } = useFetchGifs(category);

	return (
		<>
			{isLoading && <h2>Cargando...</h2>}
			<h3>{category}</h3>
			<div className='card-grid'>
				{gifs.map(({ id, title, url }) => (
					<GifGridImage key={id} title={title} url={url} />
				))}
			</div>
		</>
	);
};

GifGrid.propTypes = {
	category: PropTypes.string.isRequired,
};

export default GifGrid;
