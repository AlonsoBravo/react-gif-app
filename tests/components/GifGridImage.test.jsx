import { render, screen } from '@testing-library/react';
import { default as GifGridImage } from '../../src/components/GifGridImage';

describe('Testing in GifGridImage component', () => {
	const title = 'Goku';
	const url = 'http://www.google.cl/';

	test('should be to match snapshot', () => {
		const { container } = render(<GifGridImage title={title} url={url} />);

		expect(container).toMatchSnapshot();
	});

	test('should be show an image with correct url and alt', () => {
		render(<GifGridImage title={title} url={url} />);

		const { src, alt } = screen.getByRole('img');

		expect(src).toBe(url);
		expect(alt).toBe(title);
	});

	test('Should be show the component title', () => {
		render(<GifGridImage title={title} url={url} />);

		expect(screen.getByText(title)).toBeTruthy();
	});
});
