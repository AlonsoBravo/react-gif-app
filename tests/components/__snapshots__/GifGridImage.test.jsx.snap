// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in GifGridImage component should be to match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="http://www.google.cl/"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
